contrib/devtools/symbol-check.py

```diff
@@ -32,7 +32,7 @@
 # See https://gcc.gnu.org/onlinedocs/libstdc++/manual/abi.html for more info.
 
 MAX_VERSIONS = {
-'GCC':       (4,3,0),
+'GCC':       (12,0,0),
```

Comment:
From https://github.com/bitcoin/bitcoin/pull/30997#pullrequestreview-2337118352:
> > It'd be good if [49b0256](https://github.com/bitcoin/bitcoin/commit/49b025636be266fa17cbb4cdb9d541c0e71f2ef8) explained why bumping to 12 is needed, as the Qt 6 supported platforms for Linux claims anything down to GCC 9 should be supported: https://doc.qt.io/qt-6/supported-platforms.html#linux-x11. I'm guessing this is just because we are using 12 to compile the bins, and could also be worked around.

All new symbols are related to [software floating-point emulation](https://github.com/gcc-mirror/gcc/commit/e42d2d2a20f2bb59928bc895ec9f46503a1b5c73).

FWIW, our code does not use [`qfloat16`](https://doc.qt.io/qt-6//qfloat16.html).

Reply:
A workaround applied.